File: ase-backend/src/main/java/de/corvinrose/ase/controller/UserController.java
package de.corvinrose.ase.controller;

import de.corvinrose.ase.model.Token;
import de.corvinrose.ase.model.User;
import de.corvinrose.ase.service.AuthService;
import de.corvinrose.ase.service.UserService;
import java.util.List;
import java.util.UUID;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.*;

@AllArgsConstructor
@RestController
@RequestMapping("/user")
public class UserController {

    private final UserService userService;
    private final AuthService authService;

    @GetMapping
    public ResponseEntity<List<User>> getAllUsers() {
        List<User> users = userService.findAllUsers();
        return new ResponseEntity<>(users, HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<User> getUsersById(@PathVariable("id") UUID id) {
        User user =
                userService
                        .findUserById(id)
                        .orElseThrow(
                                () -> {
                                    throw new IllegalArgumentException(
                                            "No User with id " + id.toString() + " exists");
                                });
        return new ResponseEntity<>(user, HttpStatus.OK);
    }

    @PostMapping("/register")
    public ResponseEntity<User> registerUser(@RequestBody User user) {
        User newUser = authService.registerUser(user);
        return new ResponseEntity<>(newUser, HttpStatus.CREATED);
    }

    @PostMapping("/login")
    public ResponseEntity<Token> loginUser(@RequestBody User user) {
        Token token = authService.loginUser(user);
        return new ResponseEntity<>(token, HttpStatus.OK);
    }

    @PostMapping("/auth")
    public ResponseEntity<User> authUser(@RequestBody Token token) {
        User user = authService.authUserWithToken(token);
        return new ResponseEntity<>(user, HttpStatus.OK);
    }

    @PutMapping("/update")
    public ResponseEntity<User> updateUser(@RequestBody User user) {
        User updatedUser = userService.updateUser(user);
        return new ResponseEntity<>(updatedUser, HttpStatus.OK);
    }

    @Transactional
    @DeleteMapping("/delete/{id}")
    public ResponseEntity<User> deleteUser(@PathVariable("id") UUID id) {
        userService.deleteUser(id);
        return new ResponseEntity<>(HttpStatus.OK);
    }
}

File: ase-backend/Port/Web/Controller/UserController.php
<?php

namespace Port\Web\Controller;

use Core\App\Provider\BaseController;
use Core\App\Service\AuthService;
use Core\App\Service\UserService;
use Core\Domain\Model\Routing\HttpStatus;
use Core\Domain\Model\Routing\ResponseEntity;
use Core\Domain\Model\UserModel;
use Exception;

class UserController extends BaseController {
    private UserService $userService;
    private AuthService $authService;

    public function __construct() {
        $this->userService = new UserService();
        $this->authService = new AuthService();
    }

    /**
     * @Get /v1/user
     * @throws Exception
     */
    public function getAllUsers(): ResponseEntity {
        return new ResponseEntity($this->userService->findAllUsers(), HttpStatus::OK());
    }

    /**
     * @Get /v1/user/{id}
     * @throws Exception
     */
    public function getUsersById(string $id): ResponseEntity {
        $user = $this->userService->findUserById($id);

        $authStatus = $this->authService->authStatus();
        if ($authStatus->userName !== $user->email) {
            unset($user->password);
            unset($user->email);
        }
        return new ResponseEntity($user, HttpStatus::OK());
    }

    /**
     * @Post /v1/user/register
     * @throws Exception
     */
    public function registerUser(): ResponseEntity {
        $requestBody = file_get_contents("php://input");
        $user = UserModel::fromJsonObject(json_decode($requestBody));
        return new ResponseEntity($this->authService->registerUser($user), HttpStatus::CREATED());
    }

    /**
     * @Post /v1/user/login
     * @throws Exception
     */
    public function loginUser(): ResponseEntity {
        $requestBody = file_get_contents("php://input");
        $user = UserModel::fromJsonObject(json_decode($requestBody));
        return new ResponseEntity($this->authService->loginUser($user), HttpStatus::OK());
    }

    /**
     * @Post /v1/user/auth
     * @throws Exception
     */
    public function authUser(): ResponseEntity {
        return new ResponseEntity($this->authService->authUser(), HttpStatus::OK());
    }

    /**
     * @Put /v1/user/update
     * @throws Exception
     */
    public function updateUser(): ResponseEntity {
        $requestBody = file_get_contents("php://input");
        $user = UserModel::fromJsonObject(json_decode($requestBody));

        $authStatus = $this->authService->authStatus();
        if ($authStatus->userName !== $user->email) {
            return $this->authService->unauthorized();
        }
        return new ResponseEntity($this->userService->updateUser($user), HttpStatus::OK());
    }

    /**
     * @Delete /v1/user/delete/{id}
     * @throws Exception
     */
    public function deleteUser(string $id): ResponseEntity {
        return $this->authService->unauthorized();
        // $this->userService->deleteUser($id);
    }
}
